redfish-core/lib/account_service.hpp

```diff
@@ -1712,31 +1749,55 @@ inline void triggerUnauthenticatedACFUpload(
                     return;
                 }
             }
+        }
+    }
 
-            sdbusplus::asio::getProperty<bool>(
-                *crow::connections::systemBus, "com.ibm.PanelApp",
-                "/com/ibm/panel_app", "com.ibm.panel", "ACFWindowActive",
-                [asyncResp, decodedAcf](const boost::system::error_code ec,
-                                        const bool isACFWindowActive) {
-                if (ec)
-                {
-                    BMCWEB_LOG_ERROR
-                        << "Failed to read ACFWindowActive property";
-                    messages::internalError(asyncResp->res);
-                    return;
-                }
+    // Allow ACF upload when D-Bus property allow_unauth_upload is true (aka
+    // Redfish property AllowUnauthACFUpload).
+    sdbusplus::asio::getProperty<bool>(
+        *crow::connections::systemBus, "xyz.openbmc_project.Settings",
+        "/xyz/openbmc_project/ibmacf/allow_unauth_upload",
+        "xyz.openbmc_project.Object.Enable", "Enabled",
+        [asyncResp, decodedAcf](const boost::system::error_code& ec,
+                                const bool allowUnauthACFUpload) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR
+                << "D-Bus response error reading allow_unauth_upload: " << ec;
+            messages::internalError(asyncResp->res);
+            return;
+        }
 
-                if (isACFWindowActive)
-                {
-                    uploadACF(asyncResp, decodedAcf);
-                    return;
-                }
-                BMCWEB_LOG_ERROR << "ACF window not active";
-                messages::insufficientPrivilege(asyncResp->res);
-                return;
-                });
+        if (allowUnauthACFUpload)
+        {
+            uploadACF(asyncResp, decodedAcf);
+            return;
         }
-    }
+
+        // Allow ACF upload when D-Bus property ACFWindowActive is true
+        // (aka OpPanel function 74).
+        sdbusplus::asio::getProperty<bool>(
+            *crow::connections::systemBus, "com.ibm.PanelApp",
+            "/com/ibm/panel_app", "com.ibm.panel", "ACFWindowActive",
+            [asyncResp, decodedAcf](const boost::system::error_code& ec1,
+                                    const bool isACFWindowActive) {
+            if (ec1)
+            {
+                BMCWEB_LOG_ERROR << "Failed to read ACFWindowActive property";
+                // The Panel app doesn't run on simulated systems.
```

Comment:
As the comment says, the panel app does not run on simulated systems.  Since we use this code on QEMU and Simics systems, I think the best action is to default to a "safe"/false value and continue.  If this was only for production code, then an internal error would be more appropriate.

Reply:
Doesn't this end up with a messages::insufficientPrivilege if it hits here?